ojdbc-provider-jackson-oson/src/main/java/oracle/jdbc/provider/oson/OsonParser.java

```diff
@@ -0,0 +1,703 @@
+ ** Works (as defined below), to deal in both
+ **
+ ** (a) the Software, and
+ ** (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
+ ** one is included with the Software (each a "Larger Work" to which the Software
+ ** is contributed by such licensors),
+ **
+ ** without restriction, including without limitation the rights to copy, create
+ ** derivative works of, display, perform, and distribute the Software and make,
+ ** use, sell, offer for sale, import, export, have made, and have sold the
+ ** Software and the Larger Work(s), and to sublicense the foregoing rights on
+ ** either these or other terms.
+ **
+ ** This license is subject to the following condition:
+ ** The above copyright notice and either this complete permission notice or at
+ ** a minimum a reference to the UPL must be included in all copies or
+ ** substantial portions of the Software.
+ **
+ ** THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ ** IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ ** FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ ** AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ ** LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ ** OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ ** SOFTWARE.
+ */
+
+package oracle.jdbc.provider.oson;
+
+import com.fasterxml.jackson.core.*;
+import com.fasterxml.jackson.core.base.ParserBase;
+import com.fasterxml.jackson.core.io.IOContext;
+import oracle.sql.json.OracleJsonParser;
+import oracle.sql.json.OracleJsonParser.Event;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.OffsetDateTime;
+import java.time.Period;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * OsonParser is a custom implementation of Jackson's ParserBase class that
+ * translates events from Oracle's OracleJsonParser into JsonToken objects
+ * compatible with the Jackson library. It used for integration
+ * of Oracle's JSON parsing with Jackson's object mapping system.
+ */
+public class OsonParser extends ParserBase {
+
+  /** Logger for debugging purposes */
+  private static final boolean DEBUG = false;
+
+  /** Logger for debugging purposes */
+  private final Logger logger = Logger.getLogger("OsonLogger");
+
+  /** The OracleJsonParser instance to parse Oracle JSON data */
+  private final OracleJsonParser parser;
+
+
+  private String fieldName;
+
+  private OracleJsonParser.Event currentEvent;
+  private OracleJsonParser.Event lastClearedEvent;
+  
+  private ObjectCodec _codec;
+
+  private static final Map<OracleJsonParser.Event, JsonToken> OSON_EVENT_TO_JSON_TOKEN = new HashMap<>();
```

Comment:
This map does not seem to be used, shouldn't it be used instead of the switch/case on nextToken()?